gc/gc.go

```diff
@@ -154,7 +157,9 @@ func Descendants(ctx context.Context, getLinks dag.GetLinks, set *cid.Set, roots
 
 	for _, c := range roots {
 		// Walk recursively walks the dag and adds the keys to the given set
-		err := dag.Walk(ctx, verifyGetLinks, c, set.Visit, dag.Concurrent())
+		err := dag.Walk(ctx, verifyGetLinks, c, func(k cid.Cid) bool {
+			return set.Visit(k.Hash())
```

Comment:
So... I should have left a comment here, walking by CIDv1 was intentional.

If we walk by multihash, someone could refer to a structured node (e.g., a DagPB node) as a "Raw" node. If they did that, we'd visit the raw node, see that it was raw, and mark it as visited. As a raw node, it doesn't _have_ any children.

However, if we then encounter the same block as a DagPB node, we'd see that we've visited it and move on. Unfortunately, we'd then GC all it's children.

Reply:
Hmmm. So we revert to cid.Set + convertion to multihashes, or do you see a different way to deal with it?

Reply:
I don't. I think the visit set will always have to be true CIDs, then we'll need to convert to multihashes for the actual cleanup phase.

Note: we'll _also_ need those CIDs for the revert.

Reply:
I have reverted the change.